main/src/ca/uwaterloo/flix/language/phase/Instances.scala

```diff
@@ -27,19 +28,154 @@ import ca.uwaterloo.flix.util.{InternalCompilerException, ParOps, Validation}
+        case Expression.Ascribe(exp, _, _, _) => visit(exp)
+        case Expression.Cast(exp, _, _, _) => visit(exp)
+        case Expression.TryCatch(exp, rules, _, _, _) => visit(exp) ++ rules.flatMap(rule => visit(rule.exp))
+        case Expression.InvokeConstructor(_, args, _, _, _) => args.flatMap(visit)
+        case Expression.InvokeMethod(_, exp, args, _, _, _) => visit(exp) ++ args.flatMap(visit)
+        case Expression.InvokeStaticMethod(_, args, _, _, _) => args.flatMap(visit)
+        case Expression.GetField(_, exp, _, _, _) => visit(exp)
+        case Expression.PutField(_, exp1, exp2, _, _, _) => visit(exp1) ++ visit(exp2)
+        case Expression.GetStaticField(_, _, _, _) => Nil
+        case Expression.PutStaticField(_, exp, _, _, _) => visit(exp)
+        case Expression.NewChannel(exp, _, _, _) => visit(exp)
+        case Expression.GetChannel(exp, _, _, _) => visit(exp)
+        case Expression.PutChannel(exp1, exp2, _, _, _) => visit(exp1) ++ visit(exp2)
+        case Expression.SelectChannel(rules, default, _, _, _) => rules.flatMap(rule => visit(rule.chan) ++ visit(rule.exp)) ++ default.toList.flatMap(visit)
+        case Expression.Spawn(exp, _, _, _) => visit(exp)
+        case Expression.Lazy(exp, _, _) => visit(exp)
+        case Expression.Force(exp, _, _, _) => visit(exp)
+        case Expression.FixpointConstraintSet(_, _, _, _) => Nil
+        case Expression.FixpointCompose(exp1, exp2, _, _, _, _) => visit(exp1) ++ (visit(exp2))
+        case Expression.FixpointSolve(exp, _, _, _, _) => visit(exp)
+        case Expression.FixpointProject(_, exp, _, _, _) => visit(exp)
+        case Expression.FixpointEntails(exp1, exp2, _, _, _) => visit(exp1) ++ visit(exp2)
+        case Expression.FixpointFold(_, exp1, exp2, exp3, _, _, _) => visit(exp1) ++ visit(exp2) ++ visit(exp3)
+      }
+
+      visit(defn0.exp)
+    }
+
+    /**
+      * Checks that all signatures in `class0` are used in laws, unless `class0` is marked `lawless`.
+      */
+    def checkLawApplication(class0: TypedAst.Class): Validation[Unit, InstanceError] = class0 match {
+      case TypedAst.Class(_, mod, _, _, _, signatures, laws, _) =>
+        if (mod.isLawless) {
+          ().toSuccess
+        } else {
+          val usedSigs = laws.flatMap(findUsedSigs)
+          Validation.traverseX(signatures) {
+            sig =>
+              if (usedSigs.contains(sig.sym)) {
+                ().toSuccess
+              } else {
+                InstanceError.UnlawfulSignature(sig.sym, sig.loc).toFailure
+              }
+          }
+        }
+    }
+
+    /**
+      * Performs validations on a single class.
+      */
+    def visitClass(class0: TypedAst.Class): Validation[Unit, InstanceError] = {
+      for {
+        _ <- checkLawfulSuperClasses(class0)
+        _ <- checkLawApplication(class0)
+      } yield ()
+    }
+
+    val results = ParOps.parMap(root.classes.values, visitClass)
+    Validation.traverseX(results)(identity)
```

Comment:
Is `traverseX` with identity the same as sequence? or is it just almost the same?

Reply:
I believe it would be the same as `sequenceX` if we had it. I guess I could add that.